core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -181,6 +182,8 @@ class GroupCoordinator(val brokerId: Int,
           responseCallback(JoinGroupResult(memberId, Errors.UNKNOWN_MEMBER_ID))
         case Some(group) =>
           group.inLock {
+            if (reason != null)
+              info(s"memberId=$memberId with groupInstanceId=$groupInstanceId is attempting to join groupId=$groupId due to: $reason")
```

Comment:
We already log a few messages when a member (re-)joins. I wonder if we should pass the reason down and use it in all the existing messages that we already have. What do you think? I need to look at this a bit more myself.

Reply:
yeah, that was my original thoughts as well. i believe you are referring to `maybePrepareRebalance(group: GroupMetadata, reason: String)`. 
methods that are referencing this are:
- `addMemberAndRebalance`
- `doSyncGroup` (NA)
- `removeMemberAndUpdateGroup` (NA)
- `updateMemberAndRebalance`
- `updateStaticMemberAndRebalance`

i've passed the reason into all parent methods. also, i've updated `updateStaticMemberAndRebalance`'s direct call to `prepareRebalance()`.